contracts/metavault/strategies/StrategyIdle.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import "../../interfaces/Idle.sol";
+import "../IConverter.sol";
+import "./BaseStrategy.sol";
+
+contract StrategyIdle is BaseStrategy {
+    address public immutable idleYieldToken;
+    address public immutable IDLE;
+    address public immutable COMP;
+    IConverter public converter;
+    uint256 public virtualPrice;
+    bool public protected;
+
+    constructor(
+        address _underlying,
+        address _idleYieldToken,
+        address _IDLE,
+        address _COMP,
+        address _converter,
+        address _controller,
+        address _vaultManager,
+        address _weth,
+        address _router
+    )
+        public
+        BaseStrategy(_controller, _vaultManager, _underlying, _weth, _router)
+    {
+        idleYieldToken = _idleYieldToken;
+        IDLE = _IDLE;
+        COMP = _COMP;
+        converter = IConverter(_converter);
+        IERC20(_IDLE).safeApprove(address(_router), type(uint256).max);
+        IERC20(_COMP).safeApprove(address(_router), type(uint256).max);
+        IERC20(_underlying).safeApprove(address(_converter), type(uint256).max);
+        IERC20(_underlying).safeApprove(_idleYieldToken, type(uint256).max);
+        virtualPrice = IIdleTokenV3_1(_idleYieldToken).tokenPrice();
+        protected = true;
+    }
+
+    modifier updateVirtualPrice() {
+        if (protected) {
+            require(virtualPrice <= IIdleTokenV3_1(idleYieldToken).tokenPrice(), "virtual price is higher than needed");
+        }
+        _;
+        virtualPrice = IIdleTokenV3_1(idleYieldToken).tokenPrice();
+    }
+
+    function setProtected(bool _protected) public onlyAuthorized {
```

Comment:
Also to add to this, is there any way we can avoid having this at all? Thinking in terms of many different strategies, it would be great if we didn't have to have any one-off management for an individual one. I know sometimes that's unavoidable but figured I'd mention.